### PR TITLE
do not log error in the syscall crate

### PR DESCRIPTION
**Pull request by yihuaf**
Related to #1974. The integration tests uses `stderr` to check if there are any errors. However, there are a number of cases where we want to mount with ENOTDIR or ENOENT error but we ignore these errors. Therefore, we do not want to log these errors. Otherwise the integration tests would be broken.

tag along: Bump rust version to 1.70 and fix some comments lint in integration test. As a side effect, this will trigger the integration validation test, so we can test the fix is actually working in CI.

### CODE REVIEW

File: crates/libcontainer/src/syscall/linux.rs
Comment (utam0k) [inline crates/libcontainer/src/syscall/linux.rs:492]: How about returning the type of error in `OK`?

**Review by utam0k — CHANGES_REQUESTED** [review]
What about tracing at the caller? It is beyond the role of the syscall mod to determine the type of error.

**Comment by yihuaf** [comment]
> What about tracing at the caller? It is beyond the role of the syscall mod to determine the type of error.

I made the change for `mount` only in this PR to unblock the CI. We can extend this to all syscalls in this mod, but I'd like to do this in another PR.

**Comment by utam0k** [comment]
>  We can extend this to all syscalls in this mod, but I'd like to do this in another PR.

I agree with you. we should do it.

File: crates/libcontainer/src/syscall/linux.rs
Comment (utam0k) [inline crates/libcontainer/src/syscall/linux.rs:493]: Is there any need to bring this tracing to the callers?

Comment (yihuaf) [reply crates/libcontainer/src/syscall/linux.rs:493]: Yes, 47bc85926f55c27cefd1925df7622a0b51910d75

**Comment by yihuaf** [comment]
@utam0k since this PR hasn't been merged, I went ahead and clean up the syscall crate as well. All the simpler wrappers no longer log error and logs only at the caller site. I kept the other more complex functions such as `pivot_root` and `close_range` alone, since it makes sense to log the error there.

**Comment by yihuaf** [comment]
@utam0k Can we get a review if you have some time? I would like to merge this since this will unblock the integration validation test. I would also ask if it is possible to merge this in as is and address any reviews in a follow up PR since the integration validation CI is failing.

**Comment by YJDoc2** [comment]
Hey @yihuaf I think utam0k is unwell, so I'll take a look at this instead. The changes look ok, and as you said, if there are any issues, we can fix them in follow up PR. The only question I  have is that this was opened because  we logged errors in `mount` calls which makes the tests fail. However, I still see that we are doing tracing::error in the mount calls' `map_err` . I think you had removed is in one of the commits, but it got re-added for some reason. Can you please correct me if I'm wrong? Otherwise this is fine, I'll approve this then so it can unblock rest of the work.

**Comment by yihuaf** [comment]
> Hey @yihuaf I think utam0k is unwell, so I'll take a look at this instead. The changes look ok, and as you said, if there are any issues, we can fix them in follow up PR. The only question I have is that this was opened because we logged errors in `mount` calls which makes the tests fail. However, I still see that we are doing tracing::error in the mount calls' `map_err` . I think you had removed it in one of the commits, but it got re-added for some reason. Can you please correct me if I'm wrong? Otherwise this is fine, I'll approve this then so it can unblock rest of the work.

@YJDoc2 Thank you for taking a look. So issue that caused the integration failure is the `syscall::mount` logging error. It is removed in this PR (https://github.com/containers/youki/pull/1973/files#diff-b7916a1b6e425d0953faba26141b604a4b71e65782a58c78dd6f59f620916dccR470). The right thing to do is logging at the caller of the `syscall::mount`. There are a few places where the `syscall::mount` will return `ENOTDIR` or `ENOENT`, but the caller ignores these error because it is expected. This happens for example in the masked_path (https://github.com/containers/youki/pull/1973/files#diff-9773882fbf80be1f21b71393425530b52a9fc276cd89e5f0910d2435ca45fbb5R217). In this case, `ENOENT` is considered successful because a non-existent path is considered to be masked already (not accessible).

I believe all the tracing add you see are at the caller of `syscall::mount`. In most cases, the mount call failure indicates an actual failure, so we log them, unless the error case is handled.

I will also add one more commit to trigger the `integration_validation` test, so we can be complete in this PR.

File: crates/libcontainer/src/rootfs/rootfs.rs
Comment (yihuaf) [inline crates/libcontainer/src/rootfs/rootfs.rs:79]: For example, this call to `syscall::mount` is inside the `rootfs` module. Here, mount call failure indicates an actual failure, so we log the error here.

**Comment by YJDoc2** [comment]
Hey @yihuaf Thanks for the detailed explanation! I think I got the actual call site and implementation mixed up originally. This looks good then, and once the CI pass, I'll approve and merge. Thanks a lot!

**Pull request merged**
